1-CONTRIBUTION-GUIDE/best-practices.md

```diff
@@ -18,13 +18,12 @@ You are currently reading the best practices.
 The following guidelines are relevant to the main deployment template and nested templates (if used).
 
 1. Template parameters should follow **camelCasing**.
-2. Minimize parameters whenever possible, this allows for a good "hello world" experience where the user doesn't have to answer a number of questions to complete a deployment.  If you can use a variable or a literal, do so.  Users who want to parameterize something will likely have the skills to do so. Only provide parameters for:
- + Things that are globally unique (e.g. website name).  These are usually endpoints that the user may need to be aware of, however in many cases a unique name can be generated automatically.
+1. Minimize parameters whenever possible, this allows for a good "hello world" experience where the user doesn't have to answer a number of questions to complete a deployment.  If you can use a variable or a literal, do so.  Users who want to parameterize something will likely have the skills to do so. Only provide parameters for:
+ + Things that are globally unique (e.g. website name).  These are usually endpoints that the user may need to be aware of, however in many cases a unique name can be generated automatically by using the [uniqueString()](https://azure.microsoft.com/en-us/documentation/articles/resource-group-template-functions/#uniquestring) template language function.
```

Comment:
Add another bullet that suggests using an output for generated names that are endpoints.  E.g. if I generate a website (or VM) name, add an output to the template that outputs the uri or fqdn that was generated).

    "outputs": {
    "vm-fqdn": {
      "type": "string",
      "value": "[reference(variables('publicIPAddressName')).dnsSettings.fqdn]"
    }
  }